kernel/macc.h

```diff
@@ -107,10 +107,8 @@ struct Macc
 		std::vector<RTLIL::State> config_bits = cell->getParam("\\CONFIG").bits;
 		int config_cursor = 0;
 
-#ifndef NDEBUG
 		int config_width = cell->getParam("\\CONFIG_WIDTH").as_int();
 		log_assert(GetSize(config_bits) >= config_width);
-#endif
```

Comment:
This would slow down NDEBUG builds because now it's performing unnecessary operations, such as looking up the `CONFIG_WIDTH` IdString.

Reply:
Correct. I had to make this change because, now that we reference the `log_assert()` expression, it fails when `NDEBUG` is enabled.
I am assuming that a optimising compiler (particularly one where `NDEBUG` is enabled) would be able to figure out that the function has no side effects and its return value is discarded, and optimise the whole thing away. I am aware that this isn't guaranteed though...

The only two alternatives I can think of are:
- wrap `log_assert()`s using `config_width` behind `NDEBUG` too
- to not cache `config_width` and call `getParam` inside every `log_assert()`

unless you can think of a better way?

Reply:
`"\\CONFIG_WIDTH"` is being converted to an IdString. That alone should have enough side effect for the compiler to keep this in. Maybe instead of changing the debug macros, a better solution would be to `#ifndef NDEBUG` the variables that are in fact unused with `NDEBUG`, as happened here originally, and in ilang_backend.cc.